api/elastic.ts

```diff
@@ -66,15 +60,29 @@ const isHeight = (term: string): boolean => /^\d+$/.test(term)
 export const getSnapshot = (es: Client) => (term: string): TaskEither<ApplicationError, Snapshot> => {
     const esSearch = (isLatest(term)
         ? getLatestQuery(ESIndex.Snapshots)
-        : (isHeight(term) ? getByHeightQuery : getByHashQuery)(ESIndex.Snapshots, term))(es)
+        : getByFieldQuery(ESIndex.Snapshots, isHeight(term) ? 'height' : 'hash', term))(es)
 
     return execute(esSearch)
 }
 
 export const getCheckpointBlock = (es: Client) => (term: string): TaskEither<ApplicationError, CheckpointBlock> =>
-    execute(getByHashQuery(ESIndex.CheckpointBlocks, term)(es))
+    execute(getByFieldQuery(ESIndex.CheckpointBlocks, 'hash', term)(es))
+
+export const getTransaction = (es: Client) => (term: string): TaskEither<ApplicationError, Transaction> =>
+    execute(getByFieldQuery(ESIndex.Transactions, 'hash', term)(es))
+
+export const getTransactionBySnapshot = (es: Client) => (term: string): TaskEither<ApplicationError, Transaction[]> => {
+    if (isHeight(term)) {
+        return pipe(
+            getSnapshot(es)(term),
+            chain(snapshot => getTransactionBySnapshot(es)(snapshot.hash))
+        )
+    }
+
+    return execute(getByFieldQuery(ESIndex.Transactions, 'snapshotHash', term, -1)(es), false)
+}
 
-const execute = (search: TransportRequestPromise<ApiResponse>) => pipe(
+const execute = (search: TransportRequestPromise<ApiResponse>, single: boolean = true) => pipe(
```

Comment:
I consider such booleans as antipattern. Especially when you see `execute(blabla, flase)`.

I would rather do either:

`execute` and `executeFirst` or sth.

Reply:
On the other hand if you have `-1` in query then what is the point on passing that flag?
It is only valuable if query returns many by design and you want to take first one 🤔